### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by jmishra01**
Add few changes to improve performance and readability.

### CODE REVIEW

**Comment by hcpchris** [comment]
Changed the base for this PR to the ``develop`` branch. I'll review the details of the PR shortly.

File: highcharts_core/global_options/shared_options.py [hunk 1]
```diff
@@ -42,7 +44,7 @@ def to_js_literal(self,
 
         as_str = prefix + options_body + ');'
 
-        if filename:
+        if filename and validate.path(filename):
```
Comment (hcpchris) [inline highcharts_core/global_options/shared_options.py:47]: Can be simplified to:
```suggestion
        if validators.path(filename, allow_empty = True):
```

File: highcharts_core/chart.py [hunk 2]
```diff
@@ -351,7 +340,7 @@ def to_js_literal(self,
         suffix = """});"""
         as_str = prefix + as_str + '\n' + suffix
 
-        if filename:
+        if filename and validators.path(filename):
```
Comment (hcpchris) [inline highcharts_core/chart.py:343]: Can probably be further simplified to:
```suggestion
        if validators.path(filename, allow_empty = True):
```
which will evaluate to ``None`` if empty (and thus not trigger the condition) or throw an exception if not a valid path-like value.

**Comment by jmishra01** [comment]
Thanks for suggestion! Suggested changes are added.

**Comment by jmishra01** [comment]
Thanks for suggestion! Suggested changes are added.

**Review by hcpchris — APPROVED** [review]
Awesome - thank you for the improvements and the PR! I'll merge this into the ``develop`` branch for these changes to be included in the next release.

**Comment by jmishra01** [comment]
Thanks!! If you have any other issues or backlogs for this project, please let me know, I'm happy to contribute more on the project.

**Pull request merged**